packages/@uppy/companion/src/server/provider/dropbox/index.js

```diff
@@ -39,77 +60,48 @@ class DropBox extends Provider {
     return 'dropbox'
   }
 
-  _userInfo ({ token }, done) {
-    this.client
+  async _userInfo ({ token }) {
+    const client = this.client
       .post('users/get_current_account')
       .options({ version: '2' })
       .auth(token)
-      .request(done)
+    return promisify(client.request.bind(client))()
   }
 
   /**
-   * Makes 2 requests in parallel - 1. to get files, 2. to get user email
-   * it then waits till both requests are done before proceeding with the callback
    *
    * @param {object} options
-   * @param {Function} done
    */
-  _list (options, done) {
-    let userInfoDone = false
-    let statsDone = false
-    let userInfo
-    let stats
-    let reqErr
-    const finishReq = () => {
-      if (reqErr) {
-        logger.error(reqErr, 'provider.dropbox.list.error')
-        done(reqErr)
-      } else {
-        stats.body.user_email = userInfo.body.email
-        done(null, this.adaptData(stats.body, options.companion))
-      }
+  async list (options) {
+    try {
+      const responses = await Promise.all([
+        this._stats(options),
+        this._userInfo(options),
+      ])
+      responses.forEach((response) => {
+        if (response.statusCode !== 200) throw this._error(null, response)
+      })
+      const [{ body: stats }, { body: { email } }] = responses
+      return adaptData(stats, email, options.companion.buildURL)
+    } catch (err) {
+      logger.error(err, 'provider.dropbox.list.error')
+      throw err
     }
-
-    this.stats(options, (err, resp) => {
-      statsDone = true
-      stats = resp
-      if (err || resp.statusCode !== 200) {
-        err = this._error(err, resp)
-      }
-      reqErr = reqErr || err
-      if (userInfoDone) {
-        finishReq()
-      }
-    })
-
-    this._userInfo(options, (err, resp) => {
-      userInfoDone = true
-      userInfo = resp
-      if (err || resp.statusCode !== 200) {
-        err = this._error(err, resp)
-      }
-
-      reqErr = reqErr || err
-      if (statsDone) {
-        finishReq()
-      }
-    })
   }
 
-  stats ({ directory, query, token }, done) {
+  async _stats ({ directory, query, token }) {
     if (query.cursor) {
-      this.client
+      const client = this.client
         .post('files/list_folder/continue')
         .options({ version: '2' })
         .auth(token)
         .json({
           cursor: query.cursor,
         })
-        .request(done)
-      return
+      return promisify(client.request.bind(client))()
```

Comment:
Intead, let's pass `promise:true` when creating the purest instance here https://github.com/transloadit/uppy/blob/7b0db7d3a0163374cfbde01e826c86ead4f0aacb/packages/@uppy/companion/src/server/provider/dropbox/index.js#L2

Reply:
not sure i follow

Reply:
`purest` supports promises out of the box, let's use that instead of promisifying their API.

Reply:
i cannot find any `promise:true` option documented anywhere. are you sure it's for purest?

Reply:
ahh, now i see, v4 uses promises by default. maybe we can try to upgrade to purest v4. or v3 uses `require('purest')({ request, promise: Promise })`

Reply:
Oh ok, I've seen the `promise` option and assumed it was a boolean. Yes, let's upgrade, they've dropped support for Node.js 10.x but have we on `3.x`.

Reply:
If we're not upgrading or migrating just yet, can we use the `promise` option?